app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt

```diff
@@ -107,69 +121,125 @@ abstract class SearchPeopleViewModel(
         }
     }
 
-    fun searchQueryChanged(searchQuery: TextFieldValue) {
-        val textQueryChanged = innerSearchPeopleState.searchQuery.text != searchQuery.text
-        // we set the state with a searchQuery, immediately to update the UI first
-        innerSearchPeopleState = state.copy(searchQuery = searchQuery)
-        if (textQueryChanged) searchQueryStateFlow.search(searchQuery.text)
-    }
+}
+
+abstract class PublicWithKnownPeopleSearchViewModel(
+    private val sendConnectionRequest: SendConnectionRequestUseCase,
+    dispatcher: DispatcherProvider,
+    navigationManager: NavigationManager
+) : KnownPeopleSearchViewModel(
+    dispatcher = dispatcher,
+    navigationManager = navigationManager
+) {
+
+    private val refreshPublicResult: MutableSharedFlow<PublicRefresh> = MutableSharedFlow(0)
+
+    protected val publicPeopleSearchQueryFlow = _searchQueryFlow
+        .combine(refreshPublicResult.onSubscription { emit(PublicRefresh()) })
+        { searchTerm, publicRefresh ->
+            searchTerm to publicRefresh
+        }.flatMapLatest { (searchTerm, publicRefresh) ->
+            flow {
+                if (publicRefresh.withProgress) {
+                    emit(ContactSearchResult.ExternalContact(SearchResultState.InProgress))
+                }
 
-    private suspend fun searchKnown(searchTerm: String) {
-        localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.InProgress)
+                emit(searchPublicPeople(searchTerm))
+            }.cancellable()
+        }
 
-        val result = withContext(dispatcher.io()) { searchKnownUsersUseCase(searchTerm) }
+    fun addContact(contact: Contact) {
+        viewModelScope.launch {
+            val userId = UserId(contact.id, contact.domain)
 
-        localContactSearchResult = when (result) {
-            is SearchResult.Success -> ContactSearchResult.InternalContact(
-                SearchResultState.Success(result.contacts)
-            )
-            else -> ContactSearchResult.InternalContact(SearchResultState.Failure(R.string.label_general_error))
+            when (sendConnectionRequest(userId)) {
+                is SendConnectionRequestResult.Failure -> {
+                    appLogger.d(("Couldn't send a connect request to user $userId"))
```

Comment:
for the IDs can we not log it as it is for security reasons 
I submitted a PR in kalium to obfuscate the id to follow our security standard , you can check it out from here 
https://github.com/wireapp/kalium/pull/858/files#diff-dce27e7389f4dcd2412c0b3e8bf1a44e2aea04e60628802669e5133f6ea1d049

Reply:
I'll add a todo for now and change it once we merge your change because I think there are more places where we need to do that